src/components/api-keys/ApiKeyItem.tsx

```diff
@@ -17,33 +17,50 @@ interface Props {
   deleteApiKey: (id: string) => void;
 }
 
-export default ({ apiKey, deleteApiKey }: Props) => {
-  const { id, name, scopes, secret } = apiKey;
+export default class ApiKeyItem extends React.Component<Props, {revealSecret: boolean}> {
+  state = { revealSecret: false };
 
-  const copy = (e: any) => {
-    e.target.previousElementSibling.select();
-    document.execCommand('Copy');
-  };
+  render() {
+    const { id, name, scopes, secret } = this.props.apiKey;
+    const { revealSecret } = this.state;
 
-  return (
-    <Card fluid>
-      <Card.Content>
-        <Card.Header style={{ display: 'flex' }}>
-          <div style={{ flexGrow: 1 }}>{name}</div>
-          <div style={{ flexShrink: 0 }}>
-            <Button negative onClick={() => deleteApiKey(id)}>Delete</Button>
-          </div>
-        </Card.Header>
-        <Card.Description>
-          <ScopesList scopes={scopes}/>
-        </Card.Description>
+    const copy = (e: any) => {
+      e.target.previousElementSibling.select();
```

Comment:
just noticed this in e-mail, this is fragile. what if the dom changes and it's no longer the previous sibling? this method has a dependency on the dom structure.

instead, use refs.
https://reactjs.org/docs/refs-and-the-dom.html